lib/pure/uri.nim

```diff
@@ -289,6 +294,33 @@ func parseUri*(uri: string, result: var Uri) =
     assert res.scheme == "https"
     assert res.hostname == "nim-lang.org"
     assert res.path == "/docs/manual.html"
+
+    res = initUri()
+
+    # Non-strict
+    parseUri("https://nim-lang\n.org\t/docs/\nalert('msg\r\n')/?query\n=\tvalue#frag\nment", res, strict=false)
+    assert res.scheme == "https"
+    assert res.hostname == "nim-lang.org"
+    assert res.path == "/docs/alert('msg')/"
+    assert res.query == "query=value"
+    assert res.anchor == "fragment"
+    res = initUri()
+
+    # Strict
+    doAssertRaises(UriParseError):
+      parseUri("https://nim-lang\n.org\t/docs/\nalert('msg\r\n')/?query\n=\tvalue#frag\nment", res, strict=true)
```

Comment:
it's implicit in the declaration that strict defaults to true, so you can remove these few lines given that you're testing below with the `# Strict by default` block